app/src/main/java/com/bogdan/codeforceswatcher/features/users/redux/reducers/UsersReducer.kt

```diff
@@ -15,12 +15,13 @@ fun usersReducer(action: Action, state: AppState): UsersState {
                 status = UsersState.Status.PENDING
             )
         }
+
         is UsersRequests.FetchUsers.Success -> {
-            newState = newState.copy(
-                status = UsersState.Status.IDLE,
-                users = action.users
-            )
+            val mapUsers = action.users.associateBy { it.id }
+            val newUsers = state.users.users.map { mapUsers[it.id] ?: it }
```

Comment:
Try to swipe users list for update and in time(while users are fetching) delete one user - he appears back after fetching(but in db he is already removed). Of course if you try to remove user second time - application will crash.

Reply:
Tricky bug 👀I think, your fix should work 👍